source/common/router/router.cc

```diff
@@ -1634,9 +1634,17 @@ void Filter::onUpstreamHeaders(uint64_t response_code, Http::ResponseHeaderMapPt
   maybeProcessOrcaLoadReport(*headers, upstream_request);
 
   if (grpc_status.has_value()) {
-    upstream_request.upstreamHost()->outlierDetector().putHttpResponseCode(grpc_to_http_status);
+    upstream_request.upstreamHost()->outlierDetector().putResult(
+        grpc_to_http_status >= enumToInt(Http::Code::InternalServerError)
```

Comment:
I think this would be easier to read if you just use `500` instead of `enumToInt(Http::Code::InternalServerError)`.